src/components/DataDimension/DataDimension.js

```diff
@@ -26,6 +27,7 @@ const DataDimension = ({
                 label: item.name,
                 isActive: item.isActive,
                 type: item.type,
+                expression: item.expression,
```

Comment:
This requires the consuming app to provide `expression` for `selectedItems`, e.g. [Data Vis](https://github.com/dhis2/data-visualizer-app/blob/c41c02fc81b7351d871a760597e9a679c6a02e06/src/modules/fields/nestedFields.js#L10) needs to add `expression` to `ITEMS`.